server/src/main/java/umm3601/Server.java

```diff
@@ -86,5 +133,45 @@ public static void main(String[] args) {
     server.exception(Exception.class, (e, ctx) -> {
       throw new InternalServerErrorResponse(e.toString());
     });
+
+    return server;
+  }
+
+  /**
+   * Configure the server and the MongoDB client to shut down gracefully.
+   *
+   * @param mongoClient The MongoDB client
+   * @param server The Javalin server instance
+   */
+  private void configureShutdowns(Javalin server) {
+    /*
+     * We want the server to shut down gracefully if we kill it
+     * or if the JVM dies for some reason.
+     */
+    Runtime.getRuntime().addShutdownHook(new Thread(server::stop));
+    /*
+     * We want to shut the `mongoClient` down if the server either
+     * fails to start, or when it's shutting down for whatever reason.
+     * Since the mongClient needs to be available throughout the
+     * life of the server, the only way to do this is to wait for
+     * these events and close it then.
+     */
+    server.events(event -> {
+      event.serverStartFailed(mongoClient::close);
+      event.serverStopped(mongoClient::close);
+    });
+  }
+
+  /**
+   * Setup routes for the server.
+   *
+   * @param server The Javalin server instance
+   */
+  private void setupRoutes(Javalin server) {
+    // Add the routes for all the implementations of `Controller` in the
```

Comment:
I'd use "each of" rather than "all" since it's a for each loop.

Reply:
Yup – that would be better. I'm not in a position to make these changes right now. We're really up against the deadlines at this point, tomorrow won't be any better, and Wednesday we fly. Can you make them and re-review the code? (Will it let you re-review is you pushed the most recent changes?) If not, I'll try to get back to this when I get home.